lib/fast_jsonapi/relationship.rb

```diff
@@ -108,6 +116,11 @@ def ids_hash_from_record_and_relationship(record, params = {})
       return unless associated_object = fetch_associated_object(record, params)
 
       if associated_object.respond_to? :map
+        # Check if collection is empty to avoid strict loading violations on empty associations
+        if associated_object.respond_to?(:empty?) && associated_object.empty?
+          return []
+        end
```

Comment:
This was not detected before this PR because empty relationships was simply not serialized.